### ISSUE TITLE
Wrapping Elements fails in Microsoft Edge, and works in Chrome

### ISSUE THREAD

**Issue by bobabarbanel** [ccampbell/mousetrap#336]
I have been using Mousetrap on Chrome with the following, including wrapping specific elements.

    var trapper = document.querySelector("#"+that.getTableId()); // a table
    Mousetrap(trapper).bind(['tab', 'shift+tab', 'escape', 'enter', 'left', 'right', 'down', 'up'], etc..

It appears that on Edge, this does not work and I have to fall back to the code which does not specify the element
    Mousetrap.bind(['tab', 'shift+tab', 'escape', 'enter', 'left', 'right', 'down', 'up'], etc..

This works in Edge just fine.
The issue is that now, if I have more than one "table" in a dashboard in one document, the keystokes and user intentions will get confused.

Have anyone found a way to scope the Mousetrap in Edge?

**Comment by ccampbell** [comment]
@bobabarbanel does this work in other browsers? I don't think every type of html element supports binding key events to it. It might only work in forms/form fields and not in tables

**Comment by bobabarbanel** [comment]
It works in Chrome for Tables.
That may be right. Perhaps Edge only works for forms. I wonder if I can
insert a form without disturbing the complex layout that is present. I am
using SVGs inside a SAPUI5 table.

On Wed, Mar 16, 2016 at 9:31 AM, Craig Campbell <notifications@github.com>
wrote:

> @bobabarbanel <https://github.com/bobabarbanel> does this work in other
> browsers? I don't think every type of html element supports binding key
> events to it. It might only work in forms/form fields and not in tables
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ccampbell/mousetrap/issues/336#issuecomment-197411763>
>



-- 

Robert M. Abarbanel
bob.abarbanel@gmail.com
+1 206.499.1443

**Comment by ccampbell** [comment]
What the code is actually doing is listening for the key event, seeing what the e.target element is for the key event then saying if that or the parent element matches the one that you have wrapped mousetrap around. 

So if it is not firing the event with the table or a child of it as the target element, it won't fire the callback in mousetrap. You can still implement scoping yourself if you create multiple mousetrap objects like:

```javascript
var mousetrap1 = new Mousetrap();
var mousetrap2 = new Mousetrap();
```

And then from there you could use the [pause](https://github.com/ccampbell/mousetrap/tree/master/plugins/pause) plugin to pause and unpause the different instances depending on where people are in your application.

**Comment by bobabarbanel** [comment]
Good idea. Thank you. I will need to re-engineer things so that dashboards (multiple tables) will somehow manage the scope using pause/unpause. I may have to use mouseover type events to control the scope of the mousetraps.

**Comment by bobabarbanel** [comment]
Oh my. Turns out the Chrome version will _not work_ with the default Mousetrap.bind(). And the Microsoft version (Edge) will not work with the scoped version Mousetrap(trapper).bind().
So -- using mouseenter and mouseleave events won't work since I still must code separately for the two browsers.

**Comment by ccampbell** [comment]
@bobabarbanel what about the suggestion of using two separate instances (without wrapping around elements)? Like I posted in the earlier comment

**Comment by bobabarbanel** [comment]
That would also involve knowing which browser.

On Wed, Mar 16, 2016 at 11:09 AM, Craig Campbell <notifications@github.com>
wrote:

> @bobabarbanel <https://github.com/bobabarbanel> what about the suggestion
> of using two separate instances (without wrapping around elements)? Like I
> posted in the earlier comment
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ccampbell/mousetrap/issues/336#issuecomment-197462644>
>



-- 

Robert M. Abarbanel
bob.abarbanel@gmail.com
+1 206.499.1443

**Comment by ccampbell** [comment]
@bobabarbanel it shouldn't. The default of each mousetrap instance is to bind the events globally to `document`. It shouldn't matter where you are in your application. You would have to pause one instance and unpause the other at different times to allow different shortcuts to get captured. 

Unless the method you are using to trigger the pause/unpause does not work across browsers, it should be fine.

**Comment by bobabarbanel** [comment]
I have found that the Chrome implementation requires the scoping to the table. The Edge implementation requires the document. 

Bob Abarbanel
+1 206-499-1443
(Pleas forgive 21st speling errors. 😀)

**Comment by ccampbell** [comment]
What I am suggesting is create two separate mousetrap instances both scoped to the `document` and switch which one is active based on some other events in your application, but if that is not possible then you might have to look for another solution.
